runtime/src/qio/qio.c

```diff
@@ -1525,6 +1525,12 @@ qioerr _qio_channel_makebuffer_unlocked(qio_channel_t* ch)
 
   qbuffer_reposition(&ch->buf, start);
 
+  // Since we are just making the buffer, make sure that the
+  // available position is to the right of the start position.
+  if( ch->av_end < start ) {
+    ch->av_end = start;
```

Comment:
Yeah, more or less. When you create a channel, you can specify where it should end. Or the end might be set if we read an EOF. The end is stored in end_pos.